src/controllers/mainController.ts

```diff
@@ -321,13 +321,19 @@ export default class MainController implements vscode.Disposable {
      */
     private async createObjectExplorerSession(connectionCredentials?: IConnectionCredentials): Promise<void> {
         let createSessionPromise = new Deferred<TreeNodeInfo>();
-        await this._objectExplorerProvider.createSession(createSessionPromise, connectionCredentials);
-        const newNode = await createSessionPromise;
-        this._objectExplorerProvider.refresh(undefined);
-        let expandSessionPromise = new Deferred<TreeNodeInfo[]>();
-        await this._objectExplorerProvider.expandNode(newNode, newNode.sessionId, expandSessionPromise);
-        await expandSessionPromise;
-        this._objectExplorerProvider.refresh(undefined);
+        try {
+            await this._objectExplorerProvider.createSession(createSessionPromise, connectionCredentials);
+            const newNode = await createSessionPromise;
+            if (newNode) {
+                this._objectExplorerProvider.refresh(undefined);
+                let expandSessionPromise = new Deferred<TreeNodeInfo[]>();
+                await this._objectExplorerProvider.expandNode(newNode, newNode.sessionId, expandSessionPromise);
+                await expandSessionPromise;
+                this._objectExplorerProvider.refresh(undefined);
+            }
+        } catch (err) {
```

Comment:
Why are you catching and rethrowing? Just let the uncaught exception bubble up into a rejected promise.